extensions/yarn/main.ts

```diff
@@ -220,9 +220,9 @@ async function checkDryRun() {
 //
 // This assumes that execution was not successful and it will automatically
 // revert to the last stored package manager files.
-async function abort(code: number) {
+async function abort(code: number | null) {
   await restoreBackup();
-  Deno.exit(code);
+  Deno.exit(code ?? -1);
```

Comment:
Why was this changed? I don't think we ever abort without a code, right? And aborting with `-1` doesn't sound like a great idea.

Reply:
The code field returned by `runSandboxed` is `Option<i32>` in Rust, which becomes `number | null` in TypeScript. And we never check it against `null`. So I think we need this case.

I believe this happens when the process is killed by a signal. I am open to other values than `-1`

Reply:
This function performs the abort, it doesn't handle it. I'm slightly confused about your comment.

Reply:
Sorry. This basic code exists in many of our extensions:

```
const output = PhylumApi.runSandboxed(...);
if(!output.success) {
  await abort(output.code);
}
```

That is the code that I had in mind when making this change and writing the above comment.

But I suppose we can avoid changing this function if we handle that at the caller:

```
const output = PhylumApi.runSandboxed(...);
if(!output.success) {
  await abort(output.code ?? -1);
}
```

I will make that change

Reply:
I still wouldn't return -1, but a proper error code instead.

Reply:
Why is `-1` not a "proper error code"? Would you prefer I use `1` so that it is in the 0-255 range? Or do you think this case is important enough to have its own documented exit code?

Reply:
I don't think it's necessary to give it its own documented code, but I think we should pick one. I'd probably pick something a bit higher than 1 since that's what we do for most of our exit codes to avoid conflicts.